src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IIfStatement.cs

```diff
@@ -869,6 +869,340 @@ private static void A(bool flag, int number)
+  Condition: 
+    IParameterReferenceOperation: flag (OperationKind.ParameterReference, Type: System.Boolean) (Syntax: 'flag')
+  WhenTrue: 
+    IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null) (Syntax: '')
+      Expression: 
+        IInvalidOperation (OperationKind.Invalid, Type: null) (Syntax: '')
+          Children(0)
+  WhenFalse: 
+    IBlockOperation (1 statements) (OperationKind.Block, Type: null) (Syntax: '{ ... }')
+      IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null) (Syntax: 'Op();')
+        Expression: 
+          IInvocationOperation ( void P.Op()) (OperationKind.Invocation, Type: System.Void) (Syntax: 'Op()')
+            Instance Receiver: 
+              IInstanceReferenceOperation (ReferenceKind: ContainingTypeInstance) (OperationKind.InstanceReference, Type: P, IsImplicit) (Syntax: 'Op')
+            Arguments(0)
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // file.cs(12,28): error CS1525: Invalid expression term 'else'
+                //         /*<bind>*/if (flag)
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "").WithArguments("else").WithLocation(12, 28),
+                // file.cs(12,28): error CS1002: ; expected
+                //         /*<bind>*/if (flag)
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, "").WithLocation(12, 28)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<IfStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact, WorkItem(27866, "https://github.com/dotnet/roslyn/issues/27866")]
+        public void IIfstatementWithIfKeywordMissingAndDoubleElseKeywordsPresent()
+        {
+            string source = @"
+using System;
+
+class P
+{
+    private void Op()
+    {
+    }
+
+    private void M()
+    {
+        /*<bind>*/{
+            else
+            {
+            }
+            else
+            {
+                Op();
+            }
+        }/*</bind>*/
+    }
+}
+";
+            string expectedOperationTree = @"
+IBlockOperation (2 statements) (OperationKind.Block, Type: null, IsInvalid) (Syntax: '{ ... }')
+  IConditionalOperation (OperationKind.Conditional, Type: null, IsInvalid) (Syntax: 'else ... }')
+    Condition: 
+      IInvalidOperation (OperationKind.Invalid, Type: null) (Syntax: '')
+        Children(0)
+    WhenTrue: 
+      IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null) (Syntax: '')
+        Expression: 
+          IInvalidOperation (OperationKind.Invalid, Type: null) (Syntax: '')
+            Children(0)
+    WhenFalse: 
+      IBlockOperation (0 statements) (OperationKind.Block, Type: null, IsInvalid) (Syntax: '{ ... }')
+  IConditionalOperation (OperationKind.Conditional, Type: null) (Syntax: 'else ... }')
```

Comment:
@dotnet/analyzer-ioperation are we ok with turning this case into a conditional with a `WhenFalse` and invalid other things, instead of say just an IInvalidExpression? I'm fine with it, but I want to make sure that the rest of us are.